src/System.Memory/src/System/SpanHelpers.byte.cs

```diff
@@ -967,6 +967,63 @@ private static int LocateFirstFoundByte(Vector<byte> match)
         }
 #endif
 
+        public static unsafe int SequenceCompareTo(ref byte first, int firstLength, ref byte second, int secondLength)
+        {
+            Debug.Assert(firstLength >= 0);
+            Debug.Assert(secondLength >= 0);
+
+            if (Unsafe.AreSame(ref first, ref second))
+                goto Equal;
```

Comment:
`return firstLength.CompareTo(secondLength);` can be right here. I do not think you are getting anything by jumping to it.

Reply:
I would say this "AreSame" will be optimization only in very rare cases, but if you see it as useful, then goto solution will generate shorter native code, so I am for leaving it there.